cmd/oci/main.go

```diff
@@ -50,5 +60,48 @@ func main() {
 	log.SetOutput(f)
 
 	log.Printf("OCI FlexVolume Driver version: %s (%s)", version, build)
-	flexvolume.ExecDriver(&driver.OCIFlexvolumeDriver{}, os.Args)
+
+	registerDrivers()
+
+	driver, err := getDriverFromArgs()
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	flexvolume.ExitWithResult(flexvolume.ExecDriver(driver, os.Args))
+}
+
+func getDriverFromArgs() (flexvolume.Driver, error) {
+	driver, err := getDriver(flexvolume.DefaultSymlinkDirectory) //Block volume is default
```

Comment:
Instead of always loading the default and then overriding you can make one call to getDriver by special casing `"oci"`:

```go
driverName := strings.TrimPrefix(path.Base(os.Args[0]), "oracle~")
if driverName == "oci" {
	driverName = flexvolume.DefaultSymlinkDirectory
}